fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAnalyzerRegistryImpl.java

```diff
@@ -100,11 +100,13 @@ public LuceneAnalyzerCombinationProvider getLuceneAnalyzerCombinationProvider(@N
 
         Map<String, AnalyzerChooser> indexAnalyzerChooserPerFieldOverride = new TreeMap<>();
         Map<String, AnalyzerChooser> queryAnalyzerChooserPerFieldOverride = new TreeMap<>();
-        String[] elements = analyzerPerFieldName.split("/");
+        final String[] elements = analyzerPerFieldName.strip().split(LuceneIndexOptions.DELIMITER_BETWEEN_ELEMENTS);
 
-        for (int i = 0; i < elements.length / 2; i++) {
-            String fieldName = elements[2 * i];
-            Pair<AnalyzerChooser, AnalyzerChooser> perFieldAnalyzerChooserPair = getAnalyzerChooser(index, elements[2 * i + 1], type);
+        for (String element : elements) {
+            final String[] keyValuePair = element.strip().split(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE);
+            final String fieldName = keyValuePair[0];
+            final String analyzerName = keyValuePair[1];
+            Pair<AnalyzerChooser, AnalyzerChooser> perFieldAnalyzerChooserPair = getAnalyzerChooser(index, analyzerName, type);
```

Comment:
As written, this parser algorithm would turn the following string:

```
   a:b,  c: d,e :f
```

Into:

```
{"a": "b", "c": " d", "e ", "f"}
```

That is, it strips out whitespace at the beginning of each key and the end of each value, but not at the end of each key and the beginning of each value. Is that intentional?

Reply:
Fixed it